data/blog/webview2.md

```diff
@@ -0,0 +1,156 @@
+    | WPF w/o WebView2 (C#)     |    25,729 ms |      6 ms |        29,536 ms |         3,417 ms |
+1. Sequental IPC Round Trip
+
+    |                              | 1,000 / avg | 10,000 / avg |
+    |------------------------------|------------:|-------------:|
+    | Electron (context isolation) | 211.9 ms / 0.21 ms | 2,400 ms / 0.24 ms |
+    | Electron (node integration)  | 165.8 ms / 0.16 ms | 1,316 ms / 0.13 ms |
+    | WebView2 (C#)                     | 612.6 ms / 0.61 ms | 6,075 ms / 0.61 ms |
+    | WebView2 (C++)                    | 529.0 ms / 0.53 ms | 5,141 ms / 0.51 ms |
+1. Parallel Burst IPC Round Trip
+
+    |                              | 1,000 / avg | 10,000 / avg |
+    | -----------------------------|------------:|-------------:|
+    | Electron (context isolation) | 414 ms / 229.8 ms | 2,021 ms /  949.4 ms  |
+    | Electron (node integration)  | 138 ms /  68.1 ms | 1,349 ms /  627.5 ms  |
+    | WebView2 (C#)                     | 604 ms / 332.5 ms | 5,408 ms / 2,713.8 ms |
+    | WebView2 (C++)                    | 497 ms / 258.3 ms | 3,832 ms / 2,157.5 ms |
+
+### A Note About IPC
+
+In Chromium the browser process acts as an IPC broker between sandboxed renderers and the rest of the system.
+While Electron allows unsandboxed render processes, many apps choose to enable the sandbox for added security.
+WebView2 always has the sandbox enabled, so for most Electron and WebView2 apps IPC can heavily impact overall performance.
+
+Even though Electron and WebView2 have a similar process models, their underlying IPC implementation is different.
+Communicating between JavaScript and C++ or C# requires [marshalling](https://en.wikipedia.org/wiki/Marshalling_(computer_science)),
+most commonly into a JSON string. This is an expensive operation and IPC bottlenecks can negatively impact performance.
+
+When sending JavaScript objects between Electron's main and renderer process,
+Electron will use [the structured clone algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm) which is [significantly faster](https://github.com/crossplatform-dev/xplat-challenges/blob/main/results.md#ipc) even when using context isolation for increased security.
+Electron also supports direct IPC between any two processes via the [MessagePorts](https://www.electronjs.org/docs/latest/performance/message-ports/) API,
+which also utilizes structured clone.
+Applications which leverage this can avoid paying the JSON-serialization tax when sending objects between processes.
+
+## Summary
+
+Electron and WebView2 have a number of differences, but don't expect much difference with respect to how they perform rendering web content.
+Ultimately an app’s architecture and JavaScript libraries/frameworks have a larger impact on memory and performance than anything else because _Chromium is Chromium_ regardless of where it is running.
```

Comment:
```suggestion
Ultimately, an app’s architecture and JavaScript libraries/frameworks have a larger impact on memory and performance than anything else because _Chromium is Chromium_ regardless of where it is running.
```